modules/server/src/main/ts/bedrock/server/Controller.ts

```diff
@@ -214,20 +218,16 @@ export const create = (stickyFirstSession: boolean, singleTimeout: number, overa
             }
             clearInterval(poller);
           } else {
-            if (session.inflight !== null && (now - session.inflight.start) > (singleTimeout + timeoutGrace)) {
-              // one test took too long
-              const elapsed = formatTime(now - session.inflight.start);
-              const message = 'Test: ' + testName(session.inflight) + ' ran too long (' + elapsed + '). Limit for an individual test is set to: ' + formatTime(singleTimeout);
+             if ((now - session.updated) > (55 * 1000)) {
+              // sticky sessions have longer before they time out, but we still can't sit around if the driver has failed
+              const message = `No updates from the browser in ${formatTime(now - session.updated)}, assuming driver has failed.`;
               reject({message, results, start, now});
               clearInterval(poller);
               timeoutError = true;
             } else if (allElapsed > overallTimeout) {
               // combined tests took too long
               let lastTest;
-              if (session.inflight !== null) {
-                const runningTime = now - session.inflight.start;
-                lastTest = 'Current test: ' + testName(session.inflight) + ' running ' + formatTime(runningTime) + '.';
-              } else if (session.previous !== null) {
+              if (session.previous !== null) {
                 const sincePrevious = now - session.previous.end;
                 lastTest = 'Previous test: ' + testName(session.previous) + ' finished ' + formatTime(sincePrevious) + ' ago.';
```

Comment:
Could we some how dump the current test here as well? Sometimes I noticed test being stuck and timing out then it hard to figure out what caused it.

Reply:
This is a whole test suite timeout, not a single test stuck timeout. The single test stuck timeout is now completely client side.

The server cannot really rely on the contents of `session` here since it's updated so infrequently now. I kept printing the previous test just so we at least have _something_ that might be useful.